sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -185,7 +185,7 @@ static void _on_sock_evt(sock_udp_t *sock, sock_async_flags_t type)
                 switch (coap_get_type(&pdu)) {
                 case COAP_TYPE_NON:
                 case COAP_TYPE_ACK:
-                    if (&memo->resp_evt_tmout.queue) {
+                    if (memo->resp_evt_tmout.queue) {
```

Comment:
llvm found this even on native, and is absolutely right. The queue member in the timeout struct already is an address.

The conditional is false when `CONFIG_GCOAP_NON_TIMEOUT` to 0 and sending a NON message type. I did test this configuration earlier, but didn't verify that the conditional was false. :-/